langchain/chains/llm_router_chain/base.py

```diff
@@ -0,0 +1,103 @@
+import re
+from types import FunctionType
+from typing import Dict, List, Any
+from pydantic import Extra, root_validator
+
+from langchain import LLMChain, BasePromptTemplate
+from langchain.chains.base import Chain
+from langchain.chains.conversation.prompt import PROMPT
+from langchain.input import get_color_mapping
+from langchain.memory import ConversationBufferWindowMemory
+from langchain.schema import BaseMemory
+
+
+class RouterChain(LLMChain):
```

Comment:
name is very general, should reflect more of the specifics (uses a vector store)